### PR TITLE
Add '.take' method

### PR DESCRIPTION
**Pull request by dykov**
-----
[View rendered readme.md](https://github.com/dykov/enumerable4j/blob/add-take-method/readme.md)

### CODE REVIEW

File: src/main/java/io/github/dgroup/enumerable4j/Enumerable.java [hunk 1]
```diff
@@ -302,4 +302,24 @@ default X next(Predicate<X> prd, X alt) {
...
+     */
+    default Enumerable<X> take(long size) {
+        final Enumerable<X> out;
+        if (size < 0) {
+            throw new IllegalArgumentException(Long.toString(size));
+        } else if (size == 0) {
+            out = new Empty<>();
+        } else {
+            out = new Linked<>(
+                this.stream().limit(size).collect(Collectors.toSet())
```
Comment (dgroup) [inline src/main/java/io/github/dgroup/enumerable4j/Enumerable.java:319]: why `.toSet()`? 
Use `.toList()` in order to support the same structure

**Review by dgroup — CHANGES_REQUESTED** [review]

**Comment by dgroup** [comment]
vacation mode on :)

File: src/main/java/io/github/dgroup/enumerable4j/Enumerable.java (hunk 1 above)
Comment (dykov) [reply src/main/java/io/github/dgroup/enumerable4j/Enumerable.java]: idk why i used `.toSet()`, was inconsiderate
thanks for assist

**Review by dgroup — APPROVED** [review]

**Comment by dgroup** [comment]
@rultor merge

**Comment by dgroup** [comment]
vacation mode on

**Pull request merged**